packages/ods/src/components/pagination/tests/controller/ods-pagination.spec.ts

```diff
@@ -0,0 +1,195 @@
+
+describe('ods-pagination controller', () => {
+  beforeEach(jest.clearAllMocks);
+
+  describe('computeActualTotalPages', () => {
+    it('should return totalPages if totalItems is undefined', () => {
+      const itemPerPage = 10;
+      const totalItems = undefined;
+      const totalPages = 5;
+
+      const result = computeActualTotalPages(itemPerPage, totalItems, totalPages);
+
+      expect(result).toBe(totalPages);
+    });
+
+    it('should return the correct number of total pages when totalItems is defined', () => {
+      const itemPerPage = 10;
+      const totalItems = 95;
+      const totalPages = 5;
+
+      const result = computeActualTotalPages(itemPerPage, totalItems, totalPages);
+
+      expect(result).toBe(Math.ceil(totalItems / itemPerPage));
+    });
+
+    it('should handle itemPerPage being 0 or undefined', () => {
+      const itemPerPage = 0;
+      const totalItems = 50;
+      const totalPages = 5;
+
+      const result = computeActualTotalPages(itemPerPage, totalItems, totalPages);
+
+      expect(result).toBe(Math.ceil(totalItems / 1));
+    });
+
+    it('should handle totalItems being 0', () => {
+      const itemPerPage = 10;
+      const totalItems = 0;
+      const totalPages = 5;
+
+      const result = computeActualTotalPages(itemPerPage, totalItems, totalPages);
+
+      expect(result).toBe(totalPages);
+    });
+
+    it('should handle large numbers of totalItems and itemPerPage correctly', () => {
+      const itemPerPage = 1000;
+      const totalItems = 100000;
+      const totalPages = 100;
+
+      const result = computeActualTotalPages(itemPerPage, totalItems, totalPages);
+
+      expect(result).toBe(Math.ceil(totalItems / itemPerPage));
+    });
+
+    it('should handle small numbers of totalItems and itemPerPage correctly', () => {
+      const itemPerPage = 2;
+      const totalItems = 5;
+      const totalPages = 3;
+
+      const result = computeActualTotalPages(itemPerPage, totalItems, totalPages);
+
+      expect(result).toBe(Math.ceil(totalItems / itemPerPage));
+    });
+  });
+
+  describe('createPageList', () => {
+    it('should return all pages active when totalPages is less than or equal to MAX_VISIBLE_ITEMS', () => {
+      const totalPages = 5;
+      const pageSelected = 1;
+
+      const result = createPageList(totalPages, pageSelected);
+
+      expect(result).toHaveLength(totalPages);
+      result.forEach((page) => expect(page.active).toBe(true));
+    });
+
+    it('should activate appropriate pages when totalPages is greater than MAX_VISIBLE_ITEMS and pageSelected is in the middle', () => {
+      const totalPages = 10;
+      const pageSelected = 5;
+
+      const result = createPageList(totalPages, pageSelected);
+
+      expect(result).toHaveLength(totalPages);
+      result.forEach((page, index) => {
+        if (index === 0 || index === totalPages - 1 || (index >= (pageSelected - 2) && index <= pageSelected)) {
```

Comment:
Do not rewrite logic in unit tests (otherwise you should test it also)
use a static call for each expected results
```
expect(result[0].active).toBe(...
expect(result[1].active).toBe(...
```

Reply:
(same for all tests where you have some logic)